metadata/src/metadata/module-metadata.js

```diff
@@ -0,0 +1,265 @@
+    if(typeof extractedType === 'object'){
+        return {
+            Map: extractedType
+        }
+    }
+    return {
+        Plain: extractedType
+    };
+}
+
+/**
+ * Extract type from type object
+ * @param type type object
+ */
+function _extractType(type){
+    switch(type.kind){
+        // u8[] type alias
+        case ts.SyntaxKind.ArrayType:
+            return "Vec<u8>";
+        case ts.SyntaxKind.VoidKeyword:
+            return "void";
+        // Strip "Type" suffix from the runtime types
+        default:
+            return polkadotTypes[type.typeName.escapedText.replace("Type", "")];
+        }
+}
+
+/**
+ * Extract value of the constant
+ * @param body body of the function
+ */
+function _extractValue(body){
+    switch(body.kind){
+        case 230:
+            return body.statements[0].expression.arguments[0].text;
+        default:
+            return "0x00";
+    }
+}
+
+/**
+ * Extract parameters of the object
+ * @param params list of parameter objects
+ */
+function _extractParams(params){
+    if(params){
+        return params.map((param) => {
+            return{
+                name: param.name.escapedText,
+                type: _extractType(param.type)
+            };
+        })
+    }
+    return [];
+}
+
+/**
+ * Extract comment of the object
+ * @param jsDoc jsDoc object 
+ */
+function _extractComment(jsDoc){
+    if(!jsDoc || !jsDoc[0]){
+        return [
+            ""
+        ];
+    }
+    else if(jsDoc[0].comment && !jsDoc[0].tags){
+        return [
+            jsDoc[0].comment
+        ];
+    }
+    return jsDoc[0].tags.map(tag => tag.comment);
+}
+
+/**
+ * 
+ * @param index Module index
+ * @param node Loaded Module file
+ */
+module.exports = function generateModuleMetadata(index, node){
+    /**
+     * Template for Metadata of the module
+     */
+    let moduleMetadata = {
+        name: "",
+        storage: null,
+        calls: null,
+        events: null,
+        constants: [],
+        errors: [],
+        index: null
+    };
+
+    moduleMetadata.index = index;
+
+    /**
+     * Goes through the statements inside the file and extracts storage entries and calls
+     */
+    node.statements.forEach(obj => {
+        // Storage entries are defined inside a namespace with a corresponding name
+        if(obj.kind === ts.SyntaxKind.ModuleDeclaration){
+            const storage = _extractStorageEntries(obj);
+            moduleMetadata.storage = storage ? {
+                prefix: node.fileName,
+                items: storage
+            } : null;
+        }
+        // Calls are defined as static functions inside a class with the same name as the name of the pallet
+        else if(obj.kind === ts.SyntaxKind.ClassDeclaration){
+            moduleMetadata.name = obj.name.escapedText;
+            moduleMetadata.calls = _extractCalls(obj);
+        }
+    });
+    /**
+     * Constants for each pallet is defined in runtime directory with name pattern moduleName + prefix
```

Comment:
`Constants for each pallet is defined` -> `Constants for each pallet are defined`